pyfar/classes/audio.py

```diff
@@ -232,18 +239,31 @@ def time(self, value):
         """Set the time data."""
         # check and set the data and meta data
         data = np.atleast_2d(np.asarray(value))
-        if data.dtype.kind == "i":
-            data = data.astype("float")
-        elif data.dtype.kind != "f":
-            raise ValueError(
-                f"time data is {data.dtype}  must be int or float")
-        data = np.atleast_2d(np.asarray(value, dtype=float))
+        if self.complex:
+            data = data.astype("complex")
+            data = np.atleast_2d(np.asarray(value, dtype=complex))
+        else:
+            if data.dtype.kind == "i":
+                data = data.astype("float")
+                data = np.atleast_2d(np.asarray(value, dtype=float))
+            elif data.dtype.kind == "c":
+                raise ValueError("time data is complex, set complex "
+                                 "flag or pass real-valued data.")
+            elif data.dtype.kind != "f":
+                raise ValueError(
+                    f"time data is {data.dtype} must be int, float, or "
+                    f"complex")
+
         self._data = data
         self._n_samples = data.shape[-1]
         # setting the domain is only required for Signal. Setting it here
         # avoids the need for overloading the setter and does not harm TimeData
         self._domain = 'time'
 
+    @property
+    def complex(self):
```

Comment:
maybe we can call it `is_complex`, then its clear, that its a bool and not just complex data. We can discuss

Reply:
I think in the first version we called it "is_complex". We decided to rename it just "complex".